GliaWidgets/Sources/EntryWidget/CustomPresentationController.swift

```diff
@@ -0,0 +1,87 @@
+import UIKit
+
+final class CustomPresentationController: UIPresentationController {
+    private let height: CGFloat
+
+    private let dimmingView: UIView = {
+        let view = UIView()
+        view.backgroundColor = UIColor.black.withAlphaComponent(0.13)
+        return view
+    }()
+
+    init(
+        presentedViewController: UIViewController,
+        presenting presentingViewController: UIViewController?,
+        height: CGFloat
+    ) {
+        self.height = height
+        super.init(
+            presentedViewController: presentedViewController,
+            presenting: presentingViewController
+        )
+        let tapGesture = UITapGestureRecognizer(
+            target: self,
+            action: #selector(dimmingViewTapped)
+        )
+        dimmingView.addGestureRecognizer(tapGesture)
+    }
+
+    @objc private func dimmingViewTapped() {
+        presentedViewController.dismiss(
+            animated: true,
+            completion: nil
+        )
+    }
+
+    override var frameOfPresentedViewInContainerView: CGRect {
+        guard let containerView = containerView else { return .zero }
+
+        let topSafeAreaInset = containerView.safeAreaInsets.top
+        let yOffset = containerView.bounds.height - height - (topSafeAreaInset / 2)
+
+        return CGRect(
+            x: 0,
+            y: yOffset,
+            width: containerView.bounds.width,
+            height: height + (topSafeAreaInset / 2)
+        )
+    }
+
+    override func containerViewWillLayoutSubviews() {
+        super.containerViewWillLayoutSubviews()
+        presentedView?.layer.cornerRadius = 24
+        presentedView?.layer.masksToBounds = true
+        presentedView?.frame = frameOfPresentedViewInContainerView
+    }
+
+    override func presentationTransitionWillBegin() {
+        super.presentationTransitionWillBegin()
+
+        guard let containerView = containerView else { return }
+        dimmingView.frame = containerView.bounds
+        containerView.insertSubview(dimmingView, at: 0)
+        dimmingView.alpha = 0
+        presentedViewController.transitionCoordinator?.animate(
+            alongsideTransition: { _ in
+                self.dimmingView.alpha = 1
+            },
+            completion: nil
+        )
+    }
+
+    override func dismissalTransitionWillBegin() {
+        super.dismissalTransitionWillBegin()
+        presentedViewController.transitionCoordinator?.animate(
+            alongsideTransition: { _ in
+                self.dimmingView.alpha = 0
+            },
+            completion: nil
```

Comment:
the same

Reply:
fixed [d6cdd68](https://github.com/salemove/ios-sdk-widgets/pull/1050/commits/d6cdd68c41e5fea8e071da56a345aa320bab520f)